File: server/models/ProductModel.cs
using server.models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace server.Models
{
    public class ProductModel
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        [MaxLength(500)]
        public string Description { get; set; }

        [Required]
        public int Quantity { get; set; }

        [Required]
        public string Image { get; set; }

        [Required]
        public decimal Price { get; set; }

        // Foreign key to Category
        [ForeignKey("Category")]
        public int CategoryId { get; set; }


        // Foreign key to User
        [ForeignKey("User")]
        public int UserId { get; set; }

    }
}

File: Ecom_mobile/ecom/lib/models/product_model.dart
class Product {
  final int id;
  final String name;
  final String description;
  final int quantity;
  final String image;
  final double price;
  final int categoryId;
  final int userId;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.quantity,
    required this.image,
    required this.price,
    required this.categoryId,
    required this.userId,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      quantity: json['quantity'],
      image: json['image'],
      price: json['price'].toDouble(),
      categoryId: json['categoryId'],
      userId: json['userId'],
    );
  }
  
  String toString() {
    return 'Product(id: $id, name: $name, description: $description, price: $price, image: $image)';
  }
}
